### PR TITLE
Fixes issues with platforms w/o quarkus-bom and extension catalog metadata map merging

### LINKED ISSUE #43236: Merge extension catalog metadata properties when merging catalogs from different registries
### Description

The issue was discovered trying to setup an extension registry with a custom Quarkus platform.
The idea was to provide overrides for certain properties present in the Quarkus upstream platform catalogs for custom project creation.

Original discussion happened in https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/custom.20create.20app.20scaffolding/near/469586566

FYI @vsevel

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes https://github.com/quarkusio/quarkus/issues/43236

@ia3andy this should be it. Except I should mention `metadata/project/codestarts-data` somewhere in the docs

### CODE REVIEW

**Review by ia3andy — APPROVED** [review]
Code looks good. I haven't tested it locally but test coverage seems good!

File: docs/src/main/asciidoc/extension-codestart.adoc [hunk 1]
```diff
@@ -532,6 +532,53 @@ It is going to be merged with the other codestarts config and automatically conv
 * You can start with java and add kotlin later in another PR (create an issue so you don't forget).
 * If you have a question, ping me @ia3andy on https://quarkusio.zulipchat.com/.
 
+=== Platform codestarts data
```
Comment (aloubyansky) [inline docs/src/main/asciidoc/extension-codestart.adoc:535]: @ia3andy I added a little doc here about `metadata/project/codestart-data`

File: docs/src/main/asciidoc/extension-codestart.adoc [hunk 2]
```diff
@@ -532,6 +532,53 @@ It is going to be merged with the other codestarts config and automatically conv
...
+[source,yaml]
+----
+{
+  "id" : "org.acme.platform:acme-bom-quarkus-platform-descriptor:7.0.7:json:7.0.7",
+  "platform" : true,
+  "bom" : "org.acme.platform:acme-magic-bom::pom:7.0.7",
+  "metadata" : {
+    "project" : { <1>
+      "codestart-data" : { <2>
+        "quarkus-magic-codestart" : { <3>
```
Comment (maxandersen) [inline docs/src/main/asciidoc/extension-codestart.adoc:568]: Just to understand the scope of the data bindings here. This is linked via the code start name - nothing else; no group artifact version etc. 
Correct? 

that name isnt unique, right ? So if you have two codestarts using same name they both get this or does it get treated as an error ?

Comment (aloubyansky) [reply docs/src/main/asciidoc/extension-codestart.adoc:568]: Yes, it's the codestart name. If there are multiple codestarts with the same name **and** using the same property names that will likely lead to a problem.

**Pull request merged**